policy-controller/src/admission.rs

```diff
@@ -429,14 +445,42 @@ impl Validate<HttpRouteSpec> for Admission {
             "policy.linkerd.io HTTPRoutes must target only Server resources"
         );
 
-        // Validate the rules in this spec.
-        for rule in spec.rules.iter().flatten() {
-            ensure!(
-                !rule.has_relative_paths(),
-                "HttpRouteRules may only contain absolute paths (beginning with '/')"
-            );
-        }
+        let validate_matches = all_ok(
```

Comment:
This strikes me as overly fancy.  What's the advantage of doing this instead of just having a plain old static function `validate_rules` which calls into plain old `validate_matches` and `validate_filters` and so on?  This does incur some extra lines of code for iterating through the Vecs at each layer instead of factoring that into `all_ok`, but that seems like an okay tradeoff for being more straightforward.